src/common.rs

```diff
@@ -9,6 +10,23 @@ fn xdg_dir() -> xdg::BaseDirectories {
     xdg::BaseDirectories::with_prefix(prefix).unwrap()
 }
 
+fn data_dir(dir_name: &str) -> PathBuf {
+    // For docs.rs builds, use OUT_DIR.
+    // For other cases, use a XDG data directory.
+    // It is necessary to use OUT_DIR for docs.rs builds,
+    // as that is the only place where we can write to.
+    // The Cargo documentation recommends that build scripts
+    // place their generated files at OUT_DIR too, but we
+    // don't change that for now for normal builds.
+    if cfg!(docsrs) {
```

Comment:
I was expecting to see `if std::env::var("DOCS_RS").is_ok()` here.

Is there a reason to prefer this approach?

Reply:
This source file is used by both the build script and the application code. By checking for the feature flag we can prevent the `DOCS_RS` environment variable from interferring with the application code, and make sure that this code path is taken only on the intended environment.

Reply:
That's a good argument. Let's see what happens.